zcash_client_backend/src/scanning.rs

```diff
@@ -355,36 +355,44 @@ pub(crate) fn scan_block_with_runner<
 
     let compact_block_tx_count = block.vtx.len();
     let mut wtxs: Vec<WalletTx<K::Nf>> = vec![];
+    let mut sapling_nullifier_map = Vec::with_capacity(block.vtx.len());
     let mut sapling_note_commitments: Vec<(sapling::Node, Retention<BlockHeight>)> = vec![];
     for (tx_idx, tx) in block.vtx.into_iter().enumerate() {
         let txid = tx.txid();
+        let tx_index =
+            u16::try_from(tx.index).expect("Cannot fit more than 2^16 transactions in a block");
```

Comment:
Does the lightwalletd security model explicitly permit lightwalletd to cause a connected wallet to panic? (I know there are other cases where it can.)